ipa-core/src/protocol/basics/mul/semi_honest.rs

```diff
@@ -2,18 +2,23 @@ use async_trait::async_trait;
 
 use crate::{
     error::Error,
-    ff::Field,
+    ff::{Field, PrimeField},
     helpers::Direction,
     protocol::{
         basics::{mul::sparse::MultiplyWork, MultiplyZeroPositions},
-        context::Context,
+        context::{
+            dzkp_semi_honest::DZKPUpgraded,
```

Comment:
```suggestion
            dzkp_semi_honest::DZKPUpgraded as SemiHonestDZKPUpgraded,
```
(and update the references below)